code/modules/projectiles/guns/projectile/launchers.dm

```diff
@@ -83,3 +83,752 @@
+	name = "dud rocket"
+	desc = "A rocket round that failed to explode."
+	icon = 'icons/obj/grenade.dmi'
+	icon_state = "rocket"
+	w_class = WEIGHT_CLASS_TINY
+	slot_flags = 0
+
+/obj/item/projectile/a66mm
+	name ="66mm HE rocket"
+	desc = "A rocket with a high explosive payload."
+	icon = 'icons/obj/grenade.dmi'
+	icon_state = "SLRW-rocket"
+	alwayslog = TRUE
+	damage = 70
+	flag = "bullet"
+	hitsound = 'sound/weapons/effects/ric1.ogg'
+	hitsound_wall = 'sound/weapons/effects/ric2.ogg'
+	pass_flags = 0
+	dismemberment = 50
+	forcedodge = TRUE // Rockets don't use forcedodge anymore, but keep it anyways.
+	ricochet_chance = 0
+	range = 50
+	var/ex_range = 3 // Range of area of effect of explosion or playload.
+	var/list/rocket_stop = null // A list of types that will stop the rocket's travel.
+							
+/obj/item/projectile/a66mm/Initialize()
+	..()
+	rocket_stop = list()
+	rocket_stop += /mob/living
+	rocket_stop += /obj/mecha
+	rocket_stop += /turf/unsimulated/wall
+	rocket_stop += /turf/simulated/wall
+	rocket_stop += /obj/machinery/door/poddoor
+	rocket_stop += /obj/machinery/porta_turret
+	rocket_stop += /obj/machinery/door/airlock/vault
+	rocket_stop += /obj/machinery/door/airlock/highsecurity
+	rocket_stop += /obj/structure/shuttle
+	rocket_stop += /obj/structure/girder
+	rocket_stop += /obj/structure/window/plasmabasic
+	rocket_stop += /obj/structure/window/full/plasmabasic
+	rocket_stop += /obj/structure/window/plasmareinforced
+	rocket_stop += /obj/structure/window/full/plasmareinforced
+	
+/obj/item/projectile/a66mm/fire()
+	var/spd_multi = 1 // Speed multiplayer for rocket travel.
+	var/xd = loc.x // X location of next tile to move to.
+	var/yd = loc.y // Y location of next tile to move to.
+	transform = turn(transform, Angle)
+	
+	while(loc)
+		xd += sin(Angle)*spd_multi
+		yd += cos(Angle)*spd_multi
+		if(round(xd, 1) <= 0 || round(xd,1) >= world.maxx)
+			on_range()
+			break
+		if(round(yd, 1) <= 0 || round(yd, 1) >= world.maxy)
+			on_range()
+			break
+		new /obj/effect/temp_visual/smoke_puff(loc)
+		var/turf/next_t = locate(round(xd, 1), round(yd, 1), z)
+		if(prehit(next_t))
+			loc = next_t
+			Range()
+		else
+			break
+		sleep(max(1, speed))
+		
+/obj/item/projectile/a66mm/prehit(turf/target as turf)
+	// Check for anything that stops the rocket.
+	for(var/i in rocket_stop)
+		if(istype(target, i))
+			on_hit(target)
+			qdel(src)
+			return FALSE
+		for(var/atom/j in target.contents)
+			if(istype(j, i) && j.density)
+				on_hit(target)
+				qdel(src)
+				return FALSE
+
+	// Anything that didn't stop the rocket, destroy it.
+	var/fx_hit_flag = FALSE
+	for(var/obj/j in target.contents)
+		if(src == j)
+			continue
+		var/skip_obj = FALSE
+		for(var/i in rocket_stop)
+			if(istype(j, i))
+				skip_obj = TRUE
+				break
+		if(skip_obj)
+			continue
+		if(istype(j, /obj/machinery/door))
+			j.take_damage(j.obj_integrity)
+			fx_hit_flag = TRUE
+		if(istype(j, /obj/machinery/door/airlock/multi_tile))
+			qdel(j)
```

Comment:
There's a vertical multi tile airlock north of the bridge by the sleepers. If you hit the airlock’s north tile and destroy it, it will spawn a wreck in a different orientation. I can’t delete the wreck in a second pass since the tile is now empty. I simplified this by just deleting the entire thing.